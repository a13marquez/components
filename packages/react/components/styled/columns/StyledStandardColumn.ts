import styled from 'styled-components';
import { StandardStyles } from '../shared-styles';

interface StyledStandardColumnProps {
  readonly width?: string;
}

export const StyledStandardColumn = styled.div<StyledStandardColumnProps>`
  ${StandardStyles}
  display: flex;
  align-items: center;
  width: ${props => props.width || '20%'}
  &:nth-child(n + 2) {
    justify-content: center;
  }
`