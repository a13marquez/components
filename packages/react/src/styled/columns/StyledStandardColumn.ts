import styled from 'styled-components';
import shared from '../shared-styles';

interface StyledStandardColumnProps {
  width?: string;
}

const StyledStandardColumn = styled.div<StyledStandardColumnProps>`
  ${shared.StandardStyles}
  display: flex;
  align-items: center;
  width: ${props => props.width || '20%'};
  &:nth-child(n + 2) {
    justify-content: center;
  }
`;


export default StyledStandardColumn;