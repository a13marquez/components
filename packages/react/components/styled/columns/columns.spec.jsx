import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';

import { StyledStandardColumn } from './StyledStandardColumn';

test('It renders Styled Standard Column', () => {
  const StandardColumn = mount(<StyledStandardColumn />);
  expect(StandardColumn).toMatchSnapshot();
});

test('Styled Standard Column have the correct styles', () => {
  const StandardColumn = mount(<StyledStandardColumn />);
  expect(StandardColumn).toHaveStyleRule('display', 'flex');
  expect(StandardColumn).toHaveStyleRule('align-items', 'center');
  expect(StandardColumn).toHaveStyleRule('width', '20%');
});

test('Styled Standard Column have the correct styles when passed props', () => {
  const StandardColumn = mount(<StyledStandardColumn width="35%"/>);
  expect(StandardColumn).toHaveStyleRule('display', 'flex');
  expect(StandardColumn).toHaveStyleRule('align-items', 'center');
  expect(StandardColumn).toHaveStyleRule('width', '35%');
});

