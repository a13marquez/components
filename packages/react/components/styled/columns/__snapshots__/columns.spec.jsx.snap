// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders Styled Standard Column 1`] = `ReactWrapper {}`;
